### PR TITLE
time.Sleep removal from tests

### LINKED ISSUE #96: Get rid of time.Sleep in tests
There are some tests (especially related to P2P communication) where occasionally there are some timeouts on GitHub actions. Those tests are constructed in following manner:
* setup
* sleep and assume that setup is complete
* execute code
* sleep and assume that test is complete
* check the test assertions

Increasing sleep time is a bad idea, because it makes tests run longer then needed and still there are some timeouts in CI.
Ideally, tests assertions should be somehow triggered when execution is completed (maybe mock functions should send something to a channel, or use a WaitGroup).
Other approach is to check assertions at some short interval.

Both proposed solutions require timeout mechanism, to prevent endless wait.

### PR DESCRIPTION
**Pull request by ntsanov**
I've removed some of the time.Sleep usage in the tests that could be reliably replaced by some some event
There are still some time.Sleep invocation, especially those waiting for Node initialization. As far as I could figure out those nodes have not yet connected when p2p messages are being sent. Probably setting some event in the p2p client on connect would make it possible to determine if all nodes have established connection

Closes #96

### CODE REVIEW

**Comment by adlerjohn** [comment]
cc @tzdybal

File: node/integration_test.go [hunk 1]
```diff
@@ -197,7 +210,14 @@ func createNode(n int, aggregator bool, dalc da.DataAvailabilityLayerClient, key
...
 	app.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
-	app.On("DeliverTx", mock.Anything).Return(abci.ResponseDeliverTx{})
+	// if aggregator {
+	app.On("DeliverTx", mock.Anything).Return(abci.ResponseDeliverTx{}).Run(func(args mock.Arguments) {
+		wg.Done()
+	})
+
+	// } else {
+	// app.On("DeliverTx", mock.Anything).Return(abci.ResponseDeliverTx{})
+	// }
```
Comment (tzdybal) [inline node/integration_test.go:220]: Please remove commented-out code.

File: rpcclient/local_test.go [hunk 2]
```diff
@@ -261,7 +261,11 @@ func TestMempool2Nodes(t *testing.T) {
 	assert.Error(err)
 	assert.Nil(resp)
 
-	time.Sleep(1 * time.Second)
+	txAvailable := node2.Mempool.TxsAvailable()
+	select {
+	case <-txAvailable:
+	case <-ctx.Done():
+	}
```
Comment (tzdybal) [inline rpcclient/local_test.go:268]: :+1:

File: node/integration_test.go (hunk 1 above)
Comment (ntsanov) [reply node/integration_test.go]: Yes of course

**Review by adlerjohn — APPROVED** [review]

**Review by tzdybal — APPROVED** [review]

**Pull request merged**